### DISCUSSION TITLE
Electron to tauri system

### DISCUSSION THREAD

**Discussion by Milo123459** [tauri-apps/tauri#1223] [General]
**Is your feature request related to a problem? Please describe.**
N/A
**Describe the solution you'd like**
Not a solution, but simply a program that can convert electron apps to tauri systems.
**Describe alternatives you've considered**
N/A
**Additional context**
Not really, just maybe a cli that can convert electron apps into tauri code, would be useful for projects like VSCode, as it takes up quite a bit of memory and tauri could massively reduce that, but the devs over at VScode wouldn't be in the mood for rewriting everything again (I imagine

**Comment by nklayman** [comment]
A program that automatically converts an app would be extremely difficult, as the main process code would have to be converted from JS to Rust (at least until Tauri gets support for JS via Deno). However, we are planning on creating some tools to make the transition process easier. These will likely come after beta/stable, as they aren't of high priority yet.

**Reply by robatwilliams** [reply]
Now some time has passed, what such tools do you envision creating (or that the community could create)?

**Comment by unknown81311** [comment]
damn

**Comment by cxgreat2014** [comment]
Due to the security issues of electron, I transformed an electron project to tauri project. The biggest problem during this process is that the fs api is incompatible, and many sync api calls need to be changed to async. I spent a lot of time on refactoring it.

**Comment by betamos** [comment]
Fully automated doesn't sound feasible, but there are some aspects that could be partially automated:

- Bundling a Node runtime as a Tauri sidecar that can run the electron equivalent of the main process - relaying serialized events through the Rust binary. It would add 10s of MB to the app size. A naive implementation would likely create quite a bit of attack surface.
- Creating some form of electron-to-tauri adapter library for the electron-specific APIs. Many Electron features are not yet available in Tauri though.

**Comment by recursionbane** [comment]
Is there some guiding documentation on how Electron apps should be ported to Tauri?

**Reply by jasonm23** [reply]
Wondering the same thing... but *crickets*?

**Comment by kinthaiofficial** [comment]
Migrated a medium-sized Electron app to Tauri last year — here's the honest assessment:

**What's genuinely better in Tauri**:
- Binary size: ~3MB vs 80-150MB for Electron. Makes a big difference for distribution.
- Memory: ~30MB baseline vs 150MB+ for Electron (no bundled Chromium)
- Security model: allowlist in `tauri.conf.json` forces you to think about what OS access you're granting

**Where the migration is painful**:

```json
// Electron: easy Node.js access from renderer
// ipcRenderer.invoke('read-file', path) -> fs.readFileSync in main

// Tauri: must define each capability explicitly in Rust
// src-tauri/src/main.rs
#[tauri::command]
fn read_file(path: String) -> Result<String, String> {
    fs::read_to_string(path).map_err(|e| e.to_string())
}
```

The Rust backend requirement is the main friction point. If your Electron app has significant Node.js logic, you're rewriting it in Rust or calling out to sidecar processes.

**Practical recommendation**:
- Greenfield? Start with Tauri — the constraints are worth it
- Existing Electron app? Only migrate if binary size / memory are actual pain points for your users
- If your app is JavaScript-heavy in main process: consider wrapping existing Node.js logic as a sidecar rather than full rewrite

The webview rendering differences (OS native webview vs Chromium) also cause subtle CSS/JS behavior differences across platforms — budget time for that.
